src/android/java/AllJoynCordova.java

```diff
@@ -15,196 +16,2120 @@
+    private AJ_Object proxyObjects;
+    private AJ_Object appObjects;
+    private Timer m_pTimer = null;
+    private boolean m_bStartTimer = false;
+    private HashMap m_pMessageHandlers = new HashMap<String, String>();
+    private _AJ_Message m_pMsg = new _AJ_Message();
+
+    // Indicates if there is a callback to the web app in progress
+    // This usually means we need to stop processing messages on the loop until it is done
+    boolean m_isCallbackInProgress = false;
+    _AJ_Message m_pCallbackMessagePtr = null;
+
+    /**
+     * Sets the context of the Command. This can then be used to do things like
+     * get file paths associated with the Activity.
+     *
+     * @param cordova The context of the main Activity.
+     * @param webView The CordovaWebView Cordova is running in.
+     */
+    @Override
+    public void initialize(final CordovaInterface cordova, CordovaWebView webView)
+    {
+        super.initialize(cordova, webView);
+        Log.i(TAG, "Initialization running.");
+        alljoyn.AJ_Initialize();
+        bus = new AJ_BusAttachment();
+        proxyObjects = new AJ_Object();
+        appObjects = new AJ_Object();
+
+        // Initialize timer for msg loop
+        m_pTimer = new Timer();
+        m_pTimer.scheduleAtFixedRate
+        (
+                new TimerTask()
+                {
+                    @Override
+                    public void run()
+                    {
+                        if (!m_bStartTimer)
+                        {
+                            return;
+                        }
+
+                        AJ_Status status = alljoyn.AJ_UnmarshalMsg(bus, m_pMsg, UNMARSHAL_TIMEOUT);
+
+                        if (status == AJ_Status.AJ_OK)
+                        {
+                            final long msgId = m_pMsg.getMsgId();
+                            Log.i(TAG, "Received msgId: " + msgId);
+
+                            if (m_pMessageHandlers.containsKey(msgId))
+                            {
+                                MsgHandler handler = (MsgHandler)m_pMessageHandlers.get(msgId);
+
+                                try
+                                {
+                                    handler.callback(m_pMsg);
+                                }
+                                catch (Exception e)
+                                {
+                                    Log.i(TAG, e.toString());
+                                }
+                            }
+                            else
+                            {
+                                /*
+                                 * Pass to the built-in bus message handlers
+                                 */
+                                Log.i(TAG, "AJ_BusHandleBusMessage() msgId=" + msgId);
+                                status = alljoyn.AJ_BusHandleBusMessage(m_pMsg);
+                            }
+
+                            if (!m_isCallbackInProgress)
```

Comment:
Did you by chance try the chat sample (hosting a channel) on Android? I'm curious if you hit any issues. From the looks of it while a callback is in progress it seems that the msg loop might continue to check for messages. Might be an issue if we actually do get a message while a callback is being handled. This might be hard to hit without writing an app to make the window large enough though.